server/app/services/JiraReaderService.java

```diff
@@ -0,0 +1,264 @@
+  }
+
+  /**
+   * To request information based on status via REST API. A non-blocking call.
+   *
+   * @param status status of issue in string.
+   * @return info page encoded in JSON.
+   */
+  public CompletionStage<JsonNode> fetchIssuesForStatusByApi(String status) {
+
+    WSRequest request = ws.url("https://jira.agiledigital.com.au/rest/api/2/search")
+      .setQueryParameter("jql", "status='" + status +"'");
+    WSRequest complexRequest = request.setAuth(jiraAuth.username, jiraAuth.password, WSAuthScheme.BASIC);
+    return complexRequest.get().thenApply(WSResponse::asJson);
+  }
+
+  /**
+   * //TODO : comment this.
+   *
+   * @param response
+   * @param intent
+   * @param entity
+   * @return
+   */
+  public JsonNode read(JsonNode response, String intent, String entity) {
+    Boolean isSuccess = false;
+    System.out.println(intent + " " + entity);
+    switch (intent) {
+      case "IssueDescription":
+        isSuccess = readDescription(entity, response);
+        break;
+      case "IssueAssignee":
+        isSuccess = readAssignee(entity, response);
+        break;
+      case "IssueStatus":
+        isSuccess = readStatus(entity, response);
+        break;
+      case "AssigneeIssues":
+        isSuccess = readIssues(entity, response);
+        break;
+      case "IssuesForStatus":
+        isSuccess = readIssuesForStatus(entity, response);
+        break;
+    }
+
+    if (isSuccess) {
+      return Json.toJson(new ResponseToClient(JiraServiceProvider.REQUEST_SUCCESS, messageToReturn));
+    } else {
+      return Json.toJson(new ResponseToClient(JiraServiceProvider.REQUEST_FAILURE,
+        ConfigUtilities.getString("error-message.issue-not-found")));
+    }
+  }
+
+  /**
+   * This method reads assignee of the ticket.
+   *
+   * @param ticketNo     is the IssueID of type string which was mentioned in the query by the user.
+   * @param responseBody is the JSON object received from JIRA Rest API.
+   * @return true if success, otherwise if no such ticket exists, false.
+   */
+  private Boolean readAssignee(String ticketNo, JsonNode responseBody) {
+    if (responseBody.get("errorMessages") != null) {
+      return false;
+    } else {
+      this.messageToReturn = responseBody.get("fields").get("assignee").get("displayName").textValue()
+        + " is working on "
+        + ticketNo + ".";
+      return true;
+    }
+  }
+
+  /**
+   * This method reads description of the ticket.
+   *
+   * @param responseBody is the JSON object received from JIRA Rest API.
+   * @return true if success, otherwise if no such ticket exists, false.
+   */
+  private Boolean readDescription(String ticketNo, JsonNode responseBody) {
+    if (responseBody.get("errorMessages") != null) {
+      return false;
+    } else {
+      this.messageToReturn = hyperlinkTicketNo("Description of "
+        //  + '<' + "http://google.com" +'|' +ticketNo+  '>' // replace google by the actual ticket page
```

Comment:
I think this is the same file in another PR. Please make sure the other PR merged first, then rebase this branch's changes on top of master.